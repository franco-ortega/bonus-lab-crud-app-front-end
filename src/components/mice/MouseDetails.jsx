import React from 'react';
import PropTypes from 'prop-types';

export default function MouseDetails({ id, name, furColor, tailLength }) {
  return (
    <div>
      <p>
        ID: {id}<br />
        - Name: {name}<br />
        - Fur Color: {furColor}<br />
        - Tail Length: {tailLength} inches
      </p>
            
    </div>
  );
}

MouseDetails.propTypes = {
  id: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired,
  furColor: PropTypes.string.isRequired,
  tailLength: PropTypes.number.isRequired
};
  
