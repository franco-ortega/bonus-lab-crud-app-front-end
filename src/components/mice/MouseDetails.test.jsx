import React from 'react';
import { render, cleanup } from '@testing-library/react';
import MouseDetails from './MouseDetails';

describe('MouseDetails component', () => {
  afterEach(() => cleanup());
  it('renders MouseDetails', () => {
    const { asFragment } = render(
      <MouseDetails
        id="100"
        name="Lucia"
        furColor="gray" 
        tailLength="4" 
      />);
    expect(asFragment()).toMatchSnapshot();
  });
});
