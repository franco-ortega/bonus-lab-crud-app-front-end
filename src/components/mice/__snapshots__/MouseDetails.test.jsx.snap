// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MouseDetails component renders MouseDetails 1`] = `
<DocumentFragment>
  <div>
    <p>
      ID: 100
      <br />
      - Name: Lucia
      <br />
      - Fur Color: gray
      <br />
      - Tail Length: 4 inches
    </p>
  </div>
</DocumentFragment>
`;
